storybookjs/storybook
# Core: Fix node 18 ip support

Issue: #18019

## What I did

Running `yarn storybook` with Node 18 fails with the following issue:

<img width="489" alt="image" src="https://user-images.githubusercontent.com/1671563/170360521-38fbf532-f302-46fe-b9a8-66d691c90ce3.png">

There was an [undocumented change to the "family" attribute in os.networkInterfaces() in Node 18](https://github.com/nodejs/node/issues/42787) which made `family` become a number rather than string, as it was before. Therefore that breaks Storybook.

This PR bumps `ip` to 2.0.0. Honestly it doesn't seem like 2.0.0 has breaking changes. The real change was [this](https://github.com/indutny/node-ip/commit/4de50aec875d12b004849e11e19d6daf68b50c2d#diff-fd54b600aed6dd1ffe98402e1d4f801e8525368113f6a62ff51c81f4c7fa16f9R97) which is backwards compatible.

## How to test

To test this out I did:

1 - Access a Storybook project e.g. [MealDrop](https://github.com/yannbf/mealdrop)
2 - Set node to 18
3 - Run yarn storybook -> See failure
4 - Add resolutions to package.json
```
  "resolutions": {
    "ip": "^2.0.0"
  }
```
5 - Run yarn then yarn storybook -> See success

Would be nice to do more testing, though I'd require assistance (maybe releasing a canary?)


If your answer is yes to any of these, please make sure to include it in your PR.
